.github/workflows/test.yml

```diff
@@ -0,0 +1,157 @@
+name: Tests
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+      - stable/*
+      - release/*
+      - trying
+      - staging
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  integration-tests:
+    name: Integration tests
+    runs-on: "n1-standard-32-netssd-preempt"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - uses: stCarolas/setup-maven@v4.3
+        with:
+          maven-version: 3.8.5
+      - run: mvn -T1C -B -DskipChecks -DskipTests package
+      - run: docker build --build-arg DISTBALL=dist/target/camunda-zeebe-*.tar.gz --build-arg APP_ENV=dev -t camunda/zeebe:current-test .
+      - run: mvn -pl !:zeebe-elasticsearch-exporter -T2 -B -D skipUTs -D skipChecks -Dfailsafe.rerunFailingTestsCount=3 -Dflaky.test.reportDir=failsafe-reports -D junitThreadCount=12 -P parallel-tests,extract-flaky-tests verify
+      - name: Archive Test Results
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: Integration test results
+          path: "**/target/failsafe-reports/"
+          retention-days: 7
+  exporter-tests:
+    name: Exporter tests
+    runs-on: "n1-standard-8-netssd-preempt"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - uses: stCarolas/setup-maven@v4.3
+        with:
+          maven-version: 3.8.5
+      - run: mvn -T1C -B -DskipChecks -DskipTests package
+      - run: docker build --build-arg DISTBALL=dist/target/camunda-zeebe-*.tar.gz --build-arg APP_ENV=dev -t camunda/zeebe:current-test .
+      - run: mvn -pl :zeebe-elasticsearch-exporter -B -D skipUTs -D skipChecks -Dfailsafe.rerunFailingTestsCount=3 verify
+      - name: Archive Test Results
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: Exporter test results
+          path: "**/target/failsafe-reports/"
+          retention-days: 7
+  project-list:
+    name: List projects
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: sudo apt-get -y install jq
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - name: Get Project List
+        id: list-projects
+        run: >
+          echo "::set-output name=fast::$(mvn -pl !:zeebe-workflow-engine,!:zeebe-logstreams -Dexec.executable='echo' -Dexec.args='${project.artifactId}' exec:exec -q |  jq -cnR [inputs])"
+    outputs:
+      fast: ${{ steps.list-projects.outputs.fast }}
+  unit-tests:
+    name: Unit tests
+    needs: project-list
+    strategy:
+      fail-fast: false
+      matrix:
+        project: ${{ fromJson(needs.project-list.outputs.fast) }}
```

Comment:
👍 This is pretty clever

Reply:
Can't claim that that was my idea, it's stolen from https://tomasvotruba.com/blog/2020/11/16/how-to-make-dynamic-matrix-in-github-actions/ :sweat_smile: